src/test/scala/com/workday/warp/persistence/mysql/WarpSlickDslSpec.scala

```diff
@@ -52,6 +60,57 @@ class WarpSlickDslSpec extends WarpJUnitSpec with CorePersistenceAware {
 
     val query2 = TestExecution.filter(_.endTime isWithinPast "5 MINUTE")
     this.persistenceUtils.runWithRetries(query2.result, 5).size shouldBe 3
+
+  }
+
+
+  @Test
+  @Category(Array(classOf[UnitTest]))
+  /** Tests YEAR dsl. */
+  def returnYear(): Unit = {
+    val testExecution: TestExecutionRowLike = this.persistenceUtils.createTestExecution(methodSignature1, new Date, 1.0, 10)
+    val timeStamp: Rep[Timestamp] = testExecution.startTime
+    val query1: Rep[Int] = timeStamp year()
+    this.persistenceUtils.runWithRetries(query1.result, 5) shouldBe Year.now.getValue
+
+    val query2: Query[Rep[Int], Int, Seq] = TestExecution.map(t => t.startTime year())
+    this.persistenceUtils.runWithRetries(query2.result, 5).head shouldBe Year.now.getValue
+
+  }
+
+  @Test
+  @Category(Array(classOf[UnitTest]))
+  /** Tests DATE dsl. */
+  def returnDate(): Unit = {
+    val format: SimpleDateFormat = new java.text.SimpleDateFormat("yyyy-MM-dd")
+    val date: String = format.format(new java.util.Date())
+
+    val testExecution: TestExecutionRowLike = this.persistenceUtils.createTestExecution(methodSignature1, new Date, 1.0, 10)
+    val timeStamp: Rep[Timestamp] = testExecution.startTime
+    val query1: Rep[String] = timeStamp date()
+    this.persistenceUtils.runWithRetries(query1.result, 5) shouldBe date
+
+    val query2: Query[Rep[String], String, Seq] = TestExecution.map(t => t.startTime date())
+    this.persistenceUtils.runWithRetries(query2.result, 5).head shouldBe date
+
+  }
+
+  @Test
+  @Category(Array(classOf[UnitTest]))
+  /** Tests NOW dsl. */
+  def getCurrentTimestamp(): Unit = {
+    val testExecution: TablesLike.TestExecutionRowLike = this.persistenceUtils.createTestExecution(methodSignature1, new Date(), 1.0, 10)
+    val timeStamp: Rep[Timestamp] = testExecution.startTime
+    val query: Rep[String] = timeStamp now()
```

Comment:
I'm not sure the `now()` method  needs to be defined in the implicit `TimeStampExtension` class. I think the `now()` call can be used more generally. It doesn't really make sense to have to construct a Rep[Timestamp] just to get the current timestamp. We could just make a `HasWarpSlickDsl` companion object and put the `now()` method in there.

Reply:
are there any compiler warnings about this syntax btw? (`timestamp now()` vs `timestamp.now()`)

Reply:
I put the now function into a companion object instead. Both .now() and now() work